ldms/src/sampler/procnetdev2/procnetdev2.c

```diff
@@ -237,9 +237,10 @@ static const char *usage(struct ldmsd_plugin *self)
 		"                    whether they exist of not up to a total of MAXIFACE\n";
 }
 
-static int config(struct ldmsd_plugin *self, struct attr_value_list *kwl, struct attr_value_list *avl)
+static int config(ldmsd_plugin_handle_t self, struct attr_value_list *kwl, struct attr_value_list *avl)
 {
-	procnetdev2_t p = self->context;
+	ldmsd_cfgobj_sampler_t scfg = (ldmsd_cfgobj_sampler_t)self;
```

Comment:
This should not be this way. The plugin user should _not know_ what the handle is, and should not be able to do this cast to something that is ldmsd's internal data structures. This handle is an _opaque_ handle. Expecting the plugin author to know and cast an opaque pointer violates the entire concept of an opaque handle. That means it is no longer opaque.  See PR #1618 for how to use an opaque handle correctly.